typing/typecore.ml

```diff
@@ -6217,6 +6572,10 @@ and map_half_typed_cases
     -> ret list * partial
   = fun ?additional_checks_for_split_cases ?conts
     category env ty_arg ty_res loc caselist ~type_body ~check_if_total ->
+  let has_errors =
+    let monitor_errors = Typing_recovery.monitor_errors () in
+    not (!Clflags.typing_recovery) && !monitor_errors
```

Comment:
The doc of `Typing_recovery.monitor_errors` states:
> Returns a reference initially set to false that will be set to true
    when a type error is raised.

But here you're directly reading that reference, before an error has a chance to occur.
This make the check a few lines below useless.

Funnily enough, this is caught by one of the tests of the PR : `type_recovery_test_1` (the tests need better name btw).